search-mapper/src/main/java/org/hibernate/search/annotations/Field.java

```diff
@@ -0,0 +1,116 @@
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingStandardFieldOptionsStep;
+import org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingStep;
+
+/**
+ * Annotation used for marking a property as indexable.
+ *
+ * @author Emmanuel Bernard
+ * @author Hardy Ferentschik
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.FIELD})
+@Documented
+@Repeatable(Fields.class)
+@PropertyMapping(processor = @PropertyMappingAnnotationProcessorRef(type = Field.Processor.class))
+public @interface Field {
+
+   /**
+    * Default value for the {@link #indexNullAs} parameter. Indicates that {@code null} values should not be indexed.
+    */
+   String DO_NOT_INDEX_NULL = "__DO_NOT_INDEX_NULL__";
+
+   /**
+    * Value for the {@link #indexNullAs} parameter indicating that {@code null} values should be indexed using the null
+    * token. If no value is given for that property, the token {@code _null_} will be used.
+    */
+   String DEFAULT_NULL_TOKEN = "__DEFAULT_NULL_TOKEN__";
+
+   /**
+    * @return Returns the field name. Defaults to the JavaBean property name.
+    */
+   String name() default "";
+
+   /**
+    * @return Returns a {@code Store} enum type indicating whether the value should be stored in the document. Defaults
+    * to {@code Store.NO}.
+    */
+   Store store() default Store.NO;
+
+   /**
+    * @return Returns a {@code Index} enum defining whether the value should be indexed or not. Defaults to {@code
+    * Index.YES}.
+    */
+   Index index() default Index.YES;
+
+   /**
+    * @return Returns a {@code Analyze} enum defining whether the value should be analyzed or not. Defaults to {@code
+    * Analyze.YES}.
+    */
+   Analyze analyze() default Analyze.YES;
+
+   /**
+    * @return Returns the value to be used for indexing {@code null}. Per default {@code Field.NO_NULL_INDEXING} is
+    * returned indicating that null values are not indexed.
+    */
+   String indexNullAs() default DO_NOT_INDEX_NULL;
+
+   /**
+    * @return Whether this field should be sortable.
+    */
+   Sortable sortable() default Sortable.NO;
+
+   class Processor implements PropertyMappingAnnotationProcessor<Field> {
+
+      public static final String STANDARD_ANALYZER_NAME = "standard";
+
+      @Override
+      public void process(PropertyMappingStep mapping, Field annotation,
+                          PropertyMappingAnnotationProcessorContext context) {
+         String name = annotation.name().isEmpty() ? null : annotation.name();
+         boolean analyzed = Analyze.YES.equals(annotation.analyze());
+         boolean sortable = Sortable.YES.equals(annotation.sortable());
+
+         PropertyMappingStandardFieldOptionsStep optionsStep;
+
+         if (analyzed && !sortable) {
+            optionsStep = mapping.fullTextField(name).analyzer(STANDARD_ANALYZER_NAME);
+         } else {
```

Comment:
Unfortunately I think we cannot do that.
Actually we have a plenty of entities in the tests that have fields that are analyzed and sortable at the same time.
Because the HS5 annotation `Field` default for `analyzed` is `Analyze.YES` :(

Reply:
E.g.: see the `age` property of `AnotherGrassEater` entity.
The property type is even a numeric type, but from the annotation it results analyzed.
I really prefer the Search 6 way to handle that ;)